### PR TITLE
Make Environment.dataFiles singular

### PR DESCRIPTION
**Pull request by rjernst**
The path.data setting is now a singular string, but the method
dataFiles() that gives access to the path is still an array. This commit
renames the method and makes the return type a single Path.

relates #71205

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/env/NodeEnvironment.java [hunk 1]
```diff
@@ -195,27 +195,24 @@ public NodeLock(final Logger logger,
                         final Environment environment,
                         final CheckedFunction<Path, Boolean, IOException> pathFunction,
                         final Function<Path, Path> subPathMapping) throws IOException {
-            nodePaths = new NodePath[environment.dataFiles().length];
-            locks = new Lock[nodePaths.length];
+            nodePaths = new NodePath[1];
+            locks = new Lock[1];
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/env/NodeEnvironment.java:199]: I am assuming these are follow ups.

File: server/src/internalClusterTest/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommandIT.java [hunk 2]
```diff
@@ -566,7 +566,8 @@ public void testResolvePath() throws Exception {
         for (String nodeName : nodeNames) {
             final Path indexPath = indexPathByNodeName.get(nodeName);
             final OptionSet options = parser.parse("--dir", indexPath.toAbsolutePath().toString());
-            command.findAndProcessShardPath(options, environmentByNodeName.get(nodeName), environmentByNodeName.get(nodeName).dataFiles(),
+            command.findAndProcessShardPath(options, environmentByNodeName.get(nodeName),
+                new Path[] { environmentByNodeName.get(nodeName).dataFile() },
```
Comment (jasontedor) [inline server/src/internalClusterTest/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommandIT.java:570]: Is there potential for a follow up here to make this method parameter be single-valued?

**Review by jasontedor — APPROVED** [review]
LGTM.

File: server/src/internalClusterTest/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommandIT.java (hunk 2 above)
Comment (rjernst) [reply server/src/internalClusterTest/java/org/elasticsearch/index/shard/RemoveCorruptedShardDataCommandIT.java:570]: Yes. I'm working through essentially anything in the entire codebase that uses Path[].

File: server/src/main/java/org/elasticsearch/env/NodeEnvironment.java (hunk 1 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/env/NodeEnvironment.java:199]: Indeed.

**Pull request merged**